test/config.spec.ts

```diff
@@ -0,0 +1,26 @@
+import { DebuggerConfig } from '../src/config';
+
+describe('DebuggerConfig', () => {
+  const kernel = 'python';
+  let config: DebuggerConfig;
+
+  beforeEach(() => {
+    config = new DebuggerConfig();
+  });
+
+  describe('#getCodeId', () => {
+    it('should throw if the kernel does not have hash parameters', () => {
+      config.setTmpFileParams({ prefix: 'foo', suffix: 'bar', kernel });
+      expect(() => {
+        config.getCodeId('i = 0', kernel);
+      }).toThrow('has no hashing params');
+    });
+
+    it('should throw if the kernel does not have tmp file parameters', () => {
+      config.setHashParams({ method: 'Murmur2', seed: 'bar', kernel });
+      expect(() => {
+        config.getCodeId('i = 0', kernel);
+      }).toThrow('has no tmp file params');
```

Comment:
Does `toThrow(str)` do an `indexOf` to match that string? Because this string is a substring of the actual thrown error. I'm curious what it is passing.

Reply:
Yes, there are some examples here: https://jestjs.io/docs/en/expect#tothrowerror

But we might actually want to pass the full message to be more explicit.

Reply:
That makes sense. It is okay to leave it as is, in that case, I think.